Modules/_testinternalcapi/set.c

```diff
@@ -0,0 +1,57 @@
+#include "parts.h"
+#include "../_testcapi/util.h"  // NULLABLE, RETURN_INT
+
+#include "pycore_setobject.h"
+
+
+static PyObject *
+set_update(PyObject *self, PyObject *args)
+{
+    PyObject *set, *iterable;
+    if (!PyArg_ParseTuple(args, "OO", &set, &iterable)) {
+        return NULL;
+    }
+    NULLABLE(set);
+    NULLABLE(iterable);
+    RETURN_INT(_PySet_Update(set, iterable));
+}
+
+static PyObject *
+set_next_entry(PyObject *self, PyObject *args)
+{
+    int rc;
+    Py_ssize_t pos;
+    Py_hash_t hash;
+    PyObject *set, *item = UNINITIALIZED_PTR;
+    if (!PyArg_ParseTuple(args, "On", &set, &pos)) {
+        return NULL;
+    }
+    NULLABLE(set);
+
+    rc = _PySet_NextEntry(set, &pos, &item, &hash);
+    if (rc == 1) {
```

Comment:
What if it returns 2 or -2?
```suggestion
    if (rc != 0 && rc != -1) {
```

Reply:
Sorry, I don't understand this one. Right now it is defined as: https://github.com/python/cpython/blob/66a9b1082049855889854bfde617059499c26dd2/Objects/setobject.c#L2332-L2346

It cannot return anything except `[-1, 0, 1]`.
Do you mean that it can return something other than `[-1, 0, 1]` in the future?
I think that our test case must catch this change and be adapted if needed.

This function is not documented currently.

Reply:
The purpose of the test is to verify our assumptions (that it cannot return anything except [-1, 0, 1]). Otherwise there would not be need in tests.

In future a new return can be added in the code, or refactoring can lead to returning non-initialized variable in rare case. The wrapper will successfully return None.

Reply:
Got it, I used an assertion, which is more readable in my opinion 👍